tests/conftest.py

```diff
@@ -62,108 +160,104 @@ def bytes_helper(str, length):
+    address = w3.eth.getTransactionReceipt(tx)['contractAddress']
+    contract = w3.eth.contract(address, abi=abi, bytecode=bytecode, ContractFactoryClass=ConciseContract)
+    # Filter logs.
+    contract._logfilter = w3.eth.filter({
+        'fromBlock': w3.eth.blockNumber - 1,
+        'address': contract.address
+    })
+    return contract
 
 
 @pytest.fixture
-def utils():
-    return ethereum_utils
-
-
-@pytest.fixture
-def get_contract_from_lll(t):
-    def lll_compiler(lll):
-        lll = optimizer.optimize(LLLnode.from_list(lll))
-        byte_code = compile_lll.assembly_to_evm(compile_lll.compile_to_assembly(lll))
-        t.s.tx(to=b'', data=byte_code)
-    return lll_compiler
+def get_contract(w3):
+    def get_contract(source_code, *args, **kwargs):
+        return _get_contract(w3, source_code, *args, **kwargs)
+    return get_contract
 
 
 @pytest.fixture
-def get_contract_with_gas_estimation(chain):
-    def get_contract_with_gas_estimation(
-            source_code,
-            *args, **kwargs):
-        contract = chain.contract(source_code, language="vyper", *args, **kwargs)
-        for func_name in contract.translator.function_data:
-            set_decorator_to_contract_function(
-                chain, contract, source_code, func_name
-            )
-        return contract
+def get_contract_with_gas_estimation(w3):
+    def get_contract_with_gas_estimation(source_code, *args, **kwargs):
+        return _get_contract(w3, source_code, *args, **kwargs)
 
     return get_contract_with_gas_estimation
 
 
 @pytest.fixture
-def get_contract_with_gas_estimation_for_constants(chain):
+def get_contract_with_gas_estimation_for_constants(w3):
     def get_contract_with_gas_estimation_for_constants(
             source_code,
             *args, **kwargs):
-        abi = tester.languages['vyper'].mk_full_signature(source_code)
-        # Take out constants from the abi for the purpose of gas estimation
-        for func in abi:
-            func['constant'] = False
-        ct = tester.ContractTranslator(abi)
-        byte_code = tester.languages['vyper'].compile(source_code) + (ct.encode_constructor_arguments(kwargs['args']) if kwargs else b'')
-        address = chain.tx(to=b'', data=byte_code)
-        contract = tester.ABIContract(chain, abi, address)
-        for func_name in contract.translator.function_data:
-            set_decorator_to_contract_function(
-                chain, contract, source_code, func_name
-            )
-        return contract
+        return _get_contract(w3, source_code, *args, **kwargs)
     return get_contract_with_gas_estimation_for_constants
 
 
 @pytest.fixture
-def get_contract(chain):
-    def get_contract(source_code, *args, **kwargs):
-        return chain.contract(source_code, language="vyper", *args, **kwargs)
-    return get_contract
+def assert_tx_failed(tester):
+    def assert_tx_failed(function_to_test, exception=TransactionFailed):
```

Comment:
These would be pretty easy to implement using `contextlib.contextmanager` if you wanted to *niceness* of `pytest.raises(...)`

```python
@contextlib.contextmanager
def tx_fails(exception=...):
    try:
        yield
    except exception:
        pass
    else:
        raise AssertionError("Did not raise")
```